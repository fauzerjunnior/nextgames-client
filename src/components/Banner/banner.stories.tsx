import { Meta, StoryFn } from '@storybook/react';
import Banner, { BannerProps } from './banner.component';

export default {
  title: 'Banner',
  component: Banner,
  args: {
    img: 'https://source.unsplash.com/random/1920x1080',
    title: 'Defy death',
    subtitle: '<p>Play the new <strong>CrashLands</strong> season</p>',
    buttonLabel: 'Buy now',
    buttonLink: '/games/crashlands'
  }
} as Meta;

export const Default: StoryFn<BannerProps> = (args) => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Banner {...args} />
  </div>
);

export const WithRibbon: StoryFn<BannerProps> = (args) => {
  const withRibbonArgs = {
    ...args
  };

  return (
    <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
      <Banner
        {...withRibbonArgs}
        ribbon="20% OFF"
        ribbonColor="primary"
        ribbonSize="normal"
      />
    </div>
  );
};
