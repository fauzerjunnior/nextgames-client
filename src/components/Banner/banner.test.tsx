import { screen } from '@testing-library/react';
import Banner, { BannerProps } from './banner.component';
import { renderWithTheme } from '@/utils/tests/helpers';

describe('Banner', () => {
  const bannerPropsMock: BannerProps = {
    title: 'Defy death',
    subtitle: '<p>Play the new <strong>CrashLands</strong> season</p>',
    img: 'img/test.jpg',
    buttonLabel: 'Buy now',
    buttonLink: '/games/crashlands'
  };

  it('should render the Banner', () => {
    const { container } = renderWithTheme(<Banner {...bannerPropsMock} />);

    expect(screen.getByText(/Defy death/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the Banner with content', () => {
    renderWithTheme(<Banner {...bannerPropsMock} />);

    expect(
      screen.getByRole('heading', { name: /defy death/i })
    ).toBeInTheDocument();

    expect(
      screen.getByRole('heading', { name: /Play the new CrashLands season/i })
    );

    expect(screen.getByRole('img', { name: /defy death/i }));
  });
});
