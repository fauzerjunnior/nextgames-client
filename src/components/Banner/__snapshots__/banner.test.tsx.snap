// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render the Banner 1`] = `
.c0 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  text-decoration: none;
  cursor: pointer;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c0:hover {
  background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
}

<main
  class=""
>
  <div
    aria-label="Defy death"
    class=""
    role="img"
    src="img/test.jpg"
  />
  <div
    class=""
  >
    <h2
      class=""
    >
      Defy death
    </h2>
    <h3
      class=""
    >
      <p>
        Play the new 
        <strong>
          CrashLands
        </strong>
         season
      </p>
    </h3>
    <a
      class="c0"
      href="/games/crashlands"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</main>
`;
